classes/profile.php

```diff
@@ -159,11 +159,19 @@ private static function get_parameters(int $type): string {
      * @throws \dml_exception
      */
     public static function save(\ExcimerLog $log, int $reason, int $created, float $duration): int {
-        global $DB, $USER;
+        global $DB, $USER, $CFG;
         $flamedata = trim(str_replace("\n;", "\n", $log->formatCollapsed()));
+
+        // Remove full pathing to dirroot and only keep pathing from site root (non-issue in most sane cases).
+        $flamedata = str_replace($CFG->dirroot . DIRECTORY_SEPARATOR, '', $flamedata);
+
         $flamedatad3 = json_encode(converter::process($flamedata));
         $type = self::get_script_type();
         $parameters = self::get_parameters($type);
+
+        // If set, it will trim off the leading '/' to normalise web & cli requests.
+        $request = isset($_SERVER['PHP_SELF']) ? ltrim($_SERVER['PHP_SELF'], '/') : 'UNKNOWN';
```

Comment:
does this work when we are installed inside a folder? wonder if we used $ME it would help

Reply:
actually I think we want $SCRIPT which does the logic for us

Reply:
Tested both $ME and $SCRIPT

I think $SCRIPT is closer to what we want (example output below)
```
/admin/cli/upgrade.php
/admin/tool/excimer/profile.php
```
We don't have to do any ltrim, as both will have the `/` included in their result. However that means we will keep the `/`, assuming that's okay.

Reply:
the minimalist in me wants to ditch the / in all cases

Reply:
```suggestion
        $request = isset($SCRIPT) ? ltrim($SCRIPT, '/') : 'UNKNOWN';
```
¿Por qué no los dos? (Let's do both)

Also do you know if $SCRIPT will ever be UNKNOWN (not set)? It feels like it should always link to an existing script / file and the cases where it's "UNKNOWN" isn't made clear.

Traced it back to here: https://github.com/catalyst/moodle-tool_excimer/commit/9302a864a9491a02f5c739bf491998f486d08d1f#diff-2301d91ad2a2fdf01fbfb4be839244907712b0c1713b457e79e39a8c6d07aab4

Reply:
I think they should be rare:

https://github.com/moodle/moodle/blob/master/lib/setuplib.php#L907-L912

Reply:
I'm thinking that if the core code supports the possibility, then we should too.

Reply:
That sounds reasonable. I've kept the check and  moved the string to a class constant, and fixed up affected unit tests.